ICSharpCode.CodeConverter/CSharp/ExpressionNodeVisitor.cs

```diff
@@ -917,7 +917,17 @@ public override async Task<CSharpSyntaxNode> VisitQualifiedName(VBasic.Syntax.Qu
 
         public override async Task<CSharpSyntaxNode> VisitGenericName(VBasic.Syntax.GenericNameSyntax node)
         {
-            return SyntaxFactory.GenericName(ConvertIdentifier(node.Identifier), (TypeArgumentListSyntax) await node.TypeArgumentList.AcceptAsync(TriviaConvertingVisitor));
+            var name = SyntaxFactory.GenericName(ConvertIdentifier(node.Identifier), (TypeArgumentListSyntax) await node.TypeArgumentList.AcceptAsync(TriviaConvertingVisitor));
+
+            if (!node.Parent.IsKind(VBasic.SyntaxKind.SimpleMemberAccessExpression, VBasic.SyntaxKind.QualifiedName)) {
+                var symbol = GetSymbolInfoInDocument(node);
+                if (symbol?.ContainingSymbol != null) {
+                    var lhs = SyntaxFactory.ParseName(symbol.ContainingSymbol.ToString());
```

Comment:
I need to check what happens here if the containing symbol is generic

Reply:
Good point.

I expect this will do the wrong thing - `.ToString()` includes type parameters.

I did have some code which removed them in an earlier iteration, that would obviously do the wrong thing in this iteration though.

Reply:
An approach I've either used or thought about using elsewhere for types (which should cover the generic case) is `SyntaxFactory.ParseName(GetTypeSyntax(declaredSymbolType).ToString())` It's a big ugly but might help here and I sadly don't know a better way.

Reply:
If you need to construct the generic type name itself you can make use of CsSyntaxGenerator.GenericName which takes symbols for the parameters.

Reply:
I've added some code for this in #414, I haven't added a test case though, so I'm not 100% if it works.